components/molecules/Pagination/Pagination.cy.tsx

```diff
@@ -0,0 +1,270 @@
+        </SearchParamsContext.Provider>
+      </I18NextClientProvider>
+    );
+
+    cy.viewport(1440, 900);
+    cy.get(':nth-child(3) > [rel="next"]').should("not.exist");
+  });
+  it("renders less than 10 pages", () => {
+    const searchParams = new URLSearchParams({ page: "1" });
+    const limit = 30;
+    const offset = 0;
+    const page = 1;
+    const total = 60;
+    const paginationElements: string[] = [];
+
+    mount(
+      <I18NextClientProvider locale="en">
+        <SearchParamsContext.Provider value={searchParams}>
+          <Pagination limit={limit} offset={offset} page={page} total={total} />
+        </SearchParamsContext.Provider>
+      </I18NextClientProvider>
+    );
+
+    cy.get("ul > li").should("have.length", 2);
+    cy.get("ul > li").each(($element) =>
+      paginationElements.push($element.text())
+    );
+    cy.wrap(paginationElements).should("deep.equal", ["1", "2"]);
+  });
+  it("renders exactly 10 pages", () => {
+    const searchParams = new URLSearchParams({ page: "1" });
+    const limit = 30;
+    const offset = 0;
+    const page = 1;
+    const total = 300;
+    const paginationElements: string[] = [];
+
+    mount(
+      <I18NextClientProvider locale="en">
+        <SearchParamsContext.Provider value={searchParams}>
+          <Pagination limit={limit} offset={offset} page={page} total={total} />
+        </SearchParamsContext.Provider>
+      </I18NextClientProvider>
+    );
+
+    cy.get("ul > li").should("have.length", 10);
+    cy.get("ul > li").each(($element) =>
+      paginationElements.push($element.text())
+    );
+    cy.wrap(paginationElements).should("deep.equal", [
+      "1",
+      "2",
+      "3",
+      "4",
+      "5",
+      "6",
+      "7",
+      "8",
+      "9",
+      "10",
+    ]);
+  });
+
+  describe("renders greater than 10 pages", () => {
+    it("should have ellipses as second-to-last item when current page at the start of the page range", () => {
+      const searchParams = new URLSearchParams({ page: "1" });
+      const limit = 30;
+      const offset = 0;
+      const page = 1;
+      const total = 330;
+      const paginationElements: string[] = [];
+
+      mount(
+        <I18NextClientProvider locale="en">
+          <SearchParamsContext.Provider value={searchParams}>
+            <Pagination
+              limit={limit}
+              offset={offset}
+              page={page}
+              total={total}
+            />
+          </SearchParamsContext.Provider>
+        </I18NextClientProvider>
+      );
+
+      cy.get("ul > li").should("have.length", 10);
+      cy.get("ul > li").each(($element) =>
+        paginationElements.push($element.text())
+      );
+      cy.wrap(paginationElements).should("deep.equal", [
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+        "6",
+        "7",
+        "8",
+        "...",
+        "11",
+      ]);
+      cy.get(':nth-child(3) > [rel="prev"]').should("not.exist");
```

Comment:
suggestion: the next/prev buttons don't need to be part of these tests, their coverage can just be contained to tests that do a check for their existence and values based on inputs. A change to next/prev could end up breaking a bunch of unrelated tests and making it harder to isolate what happened.